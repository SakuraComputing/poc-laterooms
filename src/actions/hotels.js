export const addHotel = (hotel) => ({
    type: 'ADD_HOTEL',
    hotel
});

export const setHotels = (hotels) => ({
    type: 'SET_HOTELS',
    hotels
});