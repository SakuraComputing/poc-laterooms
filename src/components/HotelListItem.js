import React from 'react';

const HotelListItem = ({ name, starRating, facilities}) => {
    return (
        <div className="hotel-box">
            <h3>Hotel Name: {name}</h3>
            <div>Star Rating: {starRating}</div>
            <div>Facilities Include: {facilities.toString()}</div>
        </div>
    );
}

export default HotelListItem;
