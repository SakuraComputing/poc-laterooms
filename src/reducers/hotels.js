const hotelReducerDefaultState = [];

export default (state = hotelReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_HOTEL':
            return [
                ...state,
                action.hotel
            ];
        default:
            return state;
    }
};