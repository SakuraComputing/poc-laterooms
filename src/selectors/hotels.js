
export default (hotels, { text }) => {
    return hotels.filter((hotel) => {
        const textMatch = hotel.name.toLowerCase().includes(text.toLowerCase());
        console.log('Text Please', textMatch, hotel);
        return textMatch;
    });
};
