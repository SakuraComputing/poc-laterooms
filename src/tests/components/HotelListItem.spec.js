import React from 'react';
import { shallow } from 'enzyme';
import hotels from '../../fixtures/hotels';
import HotelListItem from '../../components/HotelListItem';

describe('Hotel List Item Compnent', () => {
    it('should render hotel list item',() => {
       const wrapper = shallow(<HotelListItem  {...hotels[0]}  />) ;
       expect(wrapper).toMatchSnapshot();
    });    
})

