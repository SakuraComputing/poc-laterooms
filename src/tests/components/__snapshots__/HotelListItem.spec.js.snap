// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotel List Item Compnent should render hotel list item 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HotelListItem
    facilities={
      Array [
        "car park",
        "pool",
      ]
    }
    name="hotelone"
    starRating={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Hotel Name: 
          hotelone
        </h3>,
        <div>
          Star Rating: 
          5
        </div>,
        <div>
          Facilities Include: 
          car park,pool
        </div>,
      ],
      "className": "hotel-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hotel Name: ",
            "hotelone",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Hotel Name: ",
          "hotelone",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Star Rating: ",
            5,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Star Rating: ",
          5,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Facilities Include: ",
            "car park,pool",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Facilities Include: ",
          "car park,pool",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Hotel Name: 
            hotelone
          </h3>,
          <div>
            Star Rating: 
            5
          </div>,
          <div>
            Facilities Include: 
            car park,pool
          </div>,
        ],
        "className": "hotel-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hotel Name: ",
              "hotelone",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Hotel Name: ",
            "hotelone",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Star Rating: ",
              5,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Star Rating: ",
            5,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Facilities Include: ",
              "car park,pool",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Facilities Include: ",
            "car park,pool",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
